### PR TITLE
Tile switching fix

### PR DESCRIPTION
**Pull request by peteruithoven**
Looks like the size_change isn't called when switching directly between tile positions using the keyboard. Possibly because the size isn't changed. But position_changed is called. 
Fixes:  #341 
Possibly also fixes: #305

### CODE REVIEW

**Comment by peteruithoven** [comment]
Like requested I tried reusing the same logic in both signal handlers. I really don't like how this looks, but I couldn't figure out a better solution.

**Comment by mtomov** [comment]
Thanks!

File: src/WindowManager.vala [hunk 1]
```diff
@@ -963,28 +963,33 @@ namespace Gala
...
-						break;
-					case Meta.SizeChange.FULLSCREEN:
-					case Meta.SizeChange.UNFULLSCREEN:
-						handle_fullscreen_window (actor.get_meta_window (), which_change);
-						break;
-				}
-
-				size_change_completed (actor);
-			});
+			}
```
Comment (tintou) [inline src/WindowManager.vala]: empty newline after closed bracket

File: src/WindowManager.vala [hunk 2]
```diff
@@ -963,28 +963,33 @@ namespace Gala
...
+					maximize (actor, new_rect.x, new_rect.y, new_rect.width, new_rect.height);
+					break;
+				case Meta.SizeChange.UNMAXIMIZE:
+					unmaximize (actor, new_rect.x, new_rect.y, new_rect.width, new_rect.height);
+					break;
+				case Meta.SizeChange.FULLSCREEN:
+				case Meta.SizeChange.UNFULLSCREEN:
+					handle_fullscreen_window (actor.get_meta_window (), which_change);
+					break;
+			}
```
Comment (tintou) [inline src/WindowManager.vala]: empty newline after closed bracket

File: src/WindowManager.vala [hunk 3]
```diff
@@ -963,28 +963,33 @@ namespace Gala
...
-				size_change_completed (actor);
-			});
+			}
+			ulong size_signal_id = 0U;
+			ulong position_signal_id = 0U;
+			size_signal_id = window.size_changed.connect (() => window_change_complete (actor, which_change, size_signal_id, position_signal_id));
+			position_signal_id = window.position_changed.connect (() => window_change_complete (actor, which_change, size_signal_id, position_signal_id));
+		}
+		
+		private void window_change_complete ( Meta.WindowActor actor, Meta.SizeChange which_change, ulong size_signal_id, ulong position_signal_id) {
```
Comment (tintou) [inline src/WindowManager.vala]:
```suggestion
		private void window_change_complete (Meta.WindowActor actor, Meta.SizeChange which_change, ulong size_signal_id, ulong position_signal_id) {
```

**Comment by peteruithoven** [comment]
Thanks for the review @tintou. I've fixed those code style issues.

**Comment by peteruithoven** [comment]
@tintou could you have a look at this again?

**Comment by peteruithoven** [comment]
@ricotz If I understand things correctly your approach isn't yet possible with valac, would it be an idea to revert back to my workaround? I could open a seperate PR with your approach, but mark it as blocked. Maybe, if available link to an relevant valac issue?

**Pull request merged**